sound/soc/sof/intel/hda-dsp.c

```diff
@@ -338,13 +338,42 @@ static int hda_dsp_send_pm_gate_ipc(struct snd_sof_dev *sdev, u32 flags)
 				  sizeof(pm_gate), &reply, sizeof(reply));
 }
 
-int hda_dsp_set_power_state(struct snd_sof_dev *sdev,
-			    enum sof_d0_substate d0_substate)
+static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
+				const struct sof_dsp_power_state *target_state)
 {
 	struct hdac_bus *bus = sof_to_bus(sdev);
-	u32 flags;
+	u32 flags = 0;
 	int ret;
-	u8 value;
+	u8 value = 0;
+
+	/*
+	 * Sanity check for illegal state transitions
+	 * The only allowed transitions are:
+	 * 1. D3 -> D0I0
+	 * 2. D0I0 -> D0I3
+	 * 3. D0I3 -> D0I0
```

Comment:
D0I0 is D0, right? I think it would be easier to read if we just use D0 here.
UPDATE: aa, the HDA-specific substate is called "D0I0", ok then.